gst/nnstreamer/tensor_query/nnstreamer_edge_common.c

```diff
@@ -0,0 +1,118 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2022 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * @file   nnstreamer_edge_internal.c
+ * @date   6 April 2022
+ * @brief  Common util functions for nnstreamer edge.
+ * @see    https://github.com/nnstreamer/nnstreamer
+ * @author Gichan Jang <gichan2.jang@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+
+#include "nnstreamer_edge_common.h"
+
+/**
+ * @brief Validate data handle.
+ */
+bool
+nns_edge_data_is_valid (nns_edge_data_h data_h)
+{
+  nns_edge_data_s *ed;
+
+  ed = (nns_edge_data_s *) data_h;
+
+  if (!NNS_EDGE_MAGIC_IS_VALID (ed))
+    return false;
+
+  return true;
+}
+
+/**
+ * @brief Create nnstreamer edge data.
+ */
+int
+nns_edge_data_create (nns_edge_data_type_e dtype, nns_edge_data_h * data_h)
+{
+  nns_edge_data_s *ed;
+
+  if (!data_h) {
+    nns_edge_loge ("Invalid param, data_h should not be null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  if (dtype < 0 || dtype >= NNS_EDGE_DATA_TYPE_MAX) {
+    nns_edge_loge ("Invalid param, given data type is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  ed = g_try_new (nns_edge_data_s, 1);
+  if (!ed) {
+    nns_edge_loge ("Failed to allocate memory for edge data.");
+    return NNS_EDGE_ERROR_OUT_OF_MEMORY;
+  }
+
+  memset (ed, 0, sizeof (nns_edge_data_s));
+  ed->magic = NNS_EDGE_MAGIC;
+  ed->dtype = dtype;
+
+  *data_h = ed;
+  return NNS_EDGE_ERROR_NONE;
+}
+
+/**
+ * @brief Destroy nnstreamer edge data.
+ */
+int
+nns_edge_data_destroy (nns_edge_data_h data_h)
+{
+  nns_edge_data_s *ed;
+  unsigned int i;
+
+  ed = (nns_edge_data_s *) data_h;
+
+  if (!nns_edge_data_is_valid (data_h)) {
+    nns_edge_loge ("Invalid param, given edge data is invalid.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  ed->magic = 0xdeaddead;
```

Comment:
Need to define the macro in `nnstreamer_edge_common.h`?

Reply:
updated

Reply:
Verified!